client/account-recovery/forgot-username/forgot-username-form/index.jsx

```diff
@@ -13,79 +14,131 @@ import Button from 'components/button';
+	firstNameUpdated = ( event ) => {
+		this.props.updatePasswordResetUserData( { firstName: event.target.value } );
 	};
 
-	firstNameUpdated = event => this.setState( { firstName: event.target.value } );
-	lastNameUpdated = event => this.setState( { lastName: event.target.value } );
-	siteUrlUpdated = event => this.setState( { siteUrl: event.target.value } );
+	lastNameUpdated = ( event ) => {
+		this.props.updatePasswordResetUserData( { lastName: event.target.value } );
+	};
 
-	render() {
+	siteUrlUpdated = ( event ) => {
+		this.props.updatePasswordResetUserData( { url: event.target.value } );
+	};
+
+	getErrorMessage = ( requestError ) => {
 		const { translate } = this.props;
-		const { isSubmitting, firstName, lastName, siteUrl } = this.state;
-		const isPrimaryButtonEnabled = firstName && lastName && siteUrl && ! isSubmitting;
+
+		if ( ! requestError ) {
+			return '';
+		}
+
+		if ( requestError.statusCode === 404 ) {
+			return translate( "We weren't able to find a user with that login information. Please try another one." );
+		}
+
+		return translate( "We've encountered some technical issues. Please try again later." );
+	};
+
+	render() {
+		const {
+			translate,
+			userData,
+			isRequesting,
+			requestError,
+		} = this.props;
+
+		const {
+			firstName,
+			lastName,
+			url,
+		} = userData;
+
+		const isPrimaryButtonEnabled = firstName && lastName && url && ! isRequesting;
 
 		return (
-			<div>
+			<Card>
 				<h2 className="forgot-username-form__title">
 					{ translate( 'Forgot your username?' ) }
 				</h2>
-				<p>{ translate( 'Enter your information to find your username' ) }</p>
-				<Card>
-					<FormLabel>
-						{ translate( 'First Name' ) }
-						<FormInput
-							className="forgot-username-form__first-name-input"
-							onChange={ this.firstNameUpdated }
-							value={ firstName }
-							disabled={ isSubmitting } />
-					</FormLabel>
-					<FormLabel>
-						{ translate( 'Last Name' ) }
-						<FormInput
-							className="forgot-username-form__last-name-input"
-							onChange={ this.lastNameUpdated }
-							value={ lastName }
-							disabled={ isSubmitting } />
-					</FormLabel>
-					<FormLabel>
-						{ translate( "Your site's URL" ) }
-						<FormInput
-							className="forgot-username-form__site-url-input"
-							onChange={ this.siteUrlUpdated }
-							value={ siteUrl }
-							disabled={ isSubmitting } />
-					</FormLabel>
-					<Button
-						className="forgot-username-form__submit-button"
-						onClick={ this.submitForm }
-						disabled={ ! isPrimaryButtonEnabled }
-						primary>
-						{ translate( 'Continue' ) }
-					</Button>
-				</Card>
-			</div>
+				<p>{ translate( 'Enter the following information instead.' ) }</p>
+				<FormLabel>
+					{ translate( 'First Name' ) }
+					<FormInput
+						className="forgot-username-form__first-name-input"
+						onChange={ this.firstNameUpdated }
+						value={ firstName || '' }
```

Comment:
I don't think we need the `firstName || ''` for these form input fields. They should treat `null`/`undefined` in the same way